### PR TITLE
Attempt to add a Project.toml file

### PR DESCRIPTION
**Pull request by omus**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,10 @@ notifications:
   email: false
 script:
   - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-  - julia -e 'Pkg.clone(pwd()); Pkg.build("Mocking"); Pkg.test("Mocking"; coverage=true)';
+  - |
+    julia -e '
+      VERSION >= v"0.7.0-DEV.3656" && using Pkg
+      VERSION >= v"0.7.0-DEV.5183" ? Pkg.instantiate() : Pkg.clone(pwd())
```
Comment (ararslan) [inline .travis.yml]: What's the difference between `Pkg.instantiate()` and `Pkg.develop(pwd())`?

Comment (omus) [reply .travis.yml]: I originally used `develop` but I believe that resulted in a UUID error. Doing a quick experiment here: https://travis-ci.org/invenia/Mocking.jl/builds/397006755

Comment (omus) [reply .travis.yml]: Results in:
```julia
ERROR: Cannot add package with the same name or uuid as the project
```

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+keywords = ["testing", "mocking"]
+author = ["Curtis Vogt"]
+license = "MIT"
+name = "Mocking"
+uuid = "78c3b35d-d492-501b-9361-3d52fe80e533"
+version = "0.5.3"
+
+[deps]
+Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
+Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
```
Comment (omus) [inline Project.toml]: Shouldn't be necessary but currently is. See https://github.com/JuliaLang/Pkg.jl/issues/427

File: .travis.yml (hunk 1 above)
Comment (ararslan) [reply .travis.yml]: I guess that only works if the package doesn't have a project file.

Comment (omus) [reply .travis.yml]: I'll run another test to see if `Pkg.instantiate()` works on packages without project files: https://travis-ci.org/invenia/Mocking.jl/builds/397024900

File: .travis.yml [hunk 3]
```diff
@@ -13,6 +13,10 @@ notifications:
...
+      Pkg.build("Mocking"); Pkg.test("Mocking"; coverage=true)'
 after_success:
   - julia -e 'cd(Pkg.dir("Mocking")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
```
Comment (omus) [inline .travis.yml]: What should I do address the `Pkg.dir` deprecation here?

File: .travis.yml (hunk 1 above)
Comment (omus) [reply .travis.yml]: Looks like `Pkg.instantiate()` works on packages with or without Project.toml files.

File: .travis.yml (hunk 3 above)
Comment (ararslan) [reply .travis.yml]: Something like
```julia
VERSION >= v"0.7.0-DEV.3656" && using Pkg
Pkg.add("Coverage")
using Coverage
dir = if VERSION >= v"0.7.0-DEV.5183"
    dirname(Base.find_package("Mocking"))
else
    Pkg.dir("Mocking")
end
cd(dir)
Codecov.submit(process_folder())
```
should work, but OH MAN is the ugly.

File: .travis.yml (hunk 1 above)
Comment (KristofferC) [reply .travis.yml]: "Idiomatic" would be something like:

```
if VERSION >= v"$PKG3_EXISTS"
    Pkg.build(); Pkg.test(;coverage = true)
else
    Pkg.clone(pwd()); Pkg.build("Mocking"); Pkg.test("Mocking"; coverage=true)'
end
```

File: .travis.yml (hunk 3 above)
Comment (KristofferC) [reply .travis.yml]: Aren't we standing in the repo when doing the `after_success`? So just remove the `Pkg.dir` completely?

**Pull request merged**